src/SessionOne.java

```diff
@@ -5,50 +5,73 @@ public static void main(String[] args) {
         Scanner input = new Scanner(System.in);
         System.out.println("Press 1, if you want to see if it's an SessionOne number");
         System.out.println("Press 2, whether you want to see if it's an Strong number");
+
         int option = input.nextInt();
+
         System.out.println("Enter a number");
         String number = String.valueOf(input.nextInt());
 
-            switch (option) {
-                case 1:
-                    isAnAmstrongNumber(number);
-                case 2:
-                    isAnStrongNumber(number);
-            }
+        switch (option) {
+            case 1:
+                isAnAmstrongNumber(number);
+            case 2:
+                isAnStrongNumber(number);
+            case 3:
+                isAPerfectNumber(Integer.valueOf(number));
         }
+    }
 
-        public static void isAnAmstrongNumber(String number) {
-            String[] digits = number.split("(?<=.)");
-            double is_amstrong = 0;
-            for (int i = 0; i < digits.length; i++) {
-                is_amstrong += Math.pow(Double.parseDouble(digits[i]), 3);
-            }
-            if (Integer.valueOf(number) == is_amstrong) {
-                System.out.println("it is an SessionOne number");
-            } else {
-                System.out.println("it is not an SessionOne number");
-            }
+    public static void isAnAmstrongNumber(String number) {
+        String[] digits = number.split("(?<=.)");
+
+        double is_amstrong = 0;
+
+        for (int i = 0; i < digits.length; i++) {
+            is_amstrong += Math.pow(Double.parseDouble(digits[i]), 3);
         }
 
-        public static void isAnStrongNumber(String number){
-            String[] digits = number.split("(?<=.)");
-
-            double isStrong = 0;
-            for (int i = 0; i < digits.length; i++) {
-                int factorial = 1;
-                int digit = Integer.valueOf(digits[i]);
-                for (int j = digit; j > 1; j=j-2) {
-                    int tempNumber = 0;
-                    tempNumber = (j * (j-1));
-                    factorial *= tempNumber;
-                }
-                isStrong +=factorial;
-            }
-            if(isStrong == Integer.valueOf(number)){
-                System.out.println("The number is a Strong number "+isStrong);
+        if (Integer.valueOf(number) == is_amstrong) {
+            System.out.println("it is an SessionOne number");
+        } else {
+            System.out.println("it is not an SessionOne number");
+        }
+    }
+
+    public static void isAnStrongNumber(String number) {
```

Comment:
```suggestion
    public static void isAStrongNumber(String number) {
```
`S` is not a vowel, should be `isAStrongNumber` instead

Reply:
I'm confused, should I commit this? but, is going to be reflected in my branch or should I do something else, I've already read the documents that you linked here, but I'm so confused.

Reply:
Yes you should commit the changes on your local branch and then push them to the remote, that will update the PR. I wrote that in my previous review, but maybe it was not that clear this might help:

![excalidraw-2020219122135 (1)](https://user-images.githubusercontent.com/10931891/74831364-6577f380-5315-11ea-9f6e-9e8122c6a4bf.png)

Try to be mindful of what you are doing, think of how git works, what does it mean I have a local branch, what does it mean that is up to date with a remote branch, don't learn to use the tool mechanically, but be aware of what you are doing, like with yoga, at the end it will be natural for you, but first you need to really understand it. If you need read out loud what is on the graphic interface when you are creating a PR, read what git says on the terminal when you do something, when you write `git status` read out loud till it makes sense to you. Please read, be aware of what you are doing means, it applies to coding too.

Reply:
Thanks for taking the time to explain this so well, and also to add all this information, it helps me so much. But I still have one question before committing the changes, because, I have included your suggestions, and also I change the project to a maven one, and create a new exercise. So, do I need to commit using commit --amend? so it goes to this PR?.

My local branch is feature/practice that is the same here, but I don't want to miss the commits that you've done, so that's why in some way I'm not sure about how to proceed.

Also, I'm going to go the workshop tomorrow, so I won't bring this laptop with me, I will be able to proceed on Saturday again. Please don't lose the faith on me. 🥇

Reply:
Including so many features in a single branch is not a good idea, try to maintain a feature based approach. That's why branches are named following this pattern `feature/feature-name` Think of it, in this way: while Carolina is working on the `feature/practice` Diana might be working on the `feature/add-maven-infrastructure` and Agustin on the `feature/remove-duplicates-from-a-list` one. All the features are going to end being merged to the `master` branch eventually (if approved of course). Why?:

* It helps to better keep track of features and potential bugs add to the codebase.
* A lot of features at the same time in the same branch make code reviews harder and annoying, it becomes difficult to keep track of what's being added and their potential effects, defeating the purpose of peer reviewing.
* It will make you life as a developer easier as well, if you mess up, you can easily go back if you have only one feature there, if you have multiple becomes more difficult. That's why you also should commit as often as you make meaningful progress on whatever you are working on.

Also you can check what commits you actually have on your branch, by running `git log` on your terminal. Or what is on the remote counterpart by checking on github: [commits of the feature/practice branch](https://github.com/c-machado/SimpliLearnSecond/commits/feature/practice) (Again explore the tool you are working with, read the interface, what says on the buttons, on the labels, on every element, click everything, get lost, but pay attention, own it!)

Remember that git commits' history is linear:

![excalidraw-202022014375](https://user-images.githubusercontent.com/10931891/74941706-caa21680-53f3-11ea-85f6-e8aeee32f6f8.png)

You might be confused because of the merging strategy you are using with your Google project, which is different, is not that one is better than the other, they are just different with their own pros and cons, but it's a good idea to understand them both. There's a task in trello regarding `rebase` merge strategy, linked with documentation about it. Please read it.

Now, in this PR you should only take care of whatever is on this PR. I would advise to remove maven configuration from this one, and the new exercise as well. And stick only to the first practice feature. Create a new branch for the maven configuration, and another one for the next exercise and so on. Get on the feature branching model mode, will make your life easier.

After you have done this, you can push the fixes you did regarding this code review, so this branch can be finally merged to master and fulfill its purpose which should be only to add the first or second exercise to the project (I am not sure). Check again the code reviews graphic, and try to understand it, to interiorize it, if there are still doubts ask again.

Reply:
You don't need to use `commit --amend` to add commits to this branch, is your local one, nothing has happened to it and only you have touched it. Again look at the commit history by using `git log`. `commit --amend` rewrites history, that's why I don't like to use it that much, it creates new commits and your local branch ends up on a divergent state from the remote counterpart, then you'll have to reconcile histories between them (I know Google loves it). More about how this works here: [git commit --amend command](https://www.atlassian.com/git/tutorials/rewriting-history)